File: 328-odd-even-linked-list/odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        oddListHead=ListNode(-1)
        oddListTail=oddListHead
        evenListHead=ListNode(-1)
        evenListTail=evenListHead
        cnt=0
        while head!=None:
            cnt=cnt+1
            if cnt%2==0:
                evenListTail.next=head
                head=head.next
                evenListTail=evenListTail.next
            else:
                oddListTail.next=head
                head=head.next
                oddListTail=oddListTail.next
        evenListTail.next=None
        oddListTail.next=evenListHead.next
        return oddListHead.next
        

File: 328-odd-even-linked-list/odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        if(head==null || head.next==null){
            return head;
        }
        ListNode oddList=new ListNode(-1);
        ListNode oddHead=oddList;
        ListNode oddTail=oddList;
        ListNode evenList=new ListNode(-1);
        ListNode evenHead=evenList;
        ListNode evenTail=evenList;
        int cnt=1;
        while(head!=null){
            if(cnt%2==0){
                evenTail.next=head;
                evenTail=evenTail.next;
            }else{
                oddTail.next=head;
                oddTail=oddTail.next;
            }
            head=head.next;
            cnt++;
        }
        evenTail.next=null;
        oddTail.next=evenHead.next;
        return oddHead.next;    
    }
}